From bfff401440c5bb72fb1ce90c804f3a5623d733b0 Mon Sep 17 00:00:00 2001
From: Connor Behan <connor.behan@gmail.com>
Date: Mon, 2 Mar 2015 16:15:16 -0500
Subject: Make it easier to find EXA

As of commit e31564e1a21e73f4d20d6471da4fc7a9b63e4062, the xserver is
more picky about the order in which headers are included. We need to
account for this in order for the compile time exa.h check to succeed.

Signed-off-by: Connor Behan <connor.behan@gmail.com>

diff --git a/configure.ac b/configure.ac
index 3cc3113..7b461c6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -121,12 +121,7 @@ CPPFLAGS="$CPPFLAGS $XORG_CFLAGS"
 AC_MSG_CHECKING([whether to enable EXA support])
 if test "x$EXA" = xyes; then
         AC_MSG_RESULT(yes)
-
-        SAVE_CPPFLAGS="$CPPFLAGS"
-        CPPFLAGS="$CPPFLAGS $XORG_CFLAGS"
-        AC_CHECK_HEADER(exa.h,
-                       [have_exa_h="yes"], [have_exa_h="no"])
-        CPPFLAGS="$SAVE_CPPFLAGS"
+        AC_CHECK_FILE(${sdkdir}/exa.h, [have_exa_h="yes"], [have_exa_h="no"])
 else
         AC_MSG_RESULT(no)
 fi
@@ -136,6 +131,7 @@ CPPFLAGS="$CPPFLAGS $XORG_CFLAGS"
 if test "x$have_exa_h" = xyes; then
         AC_MSG_CHECKING([whether EXA version is at least 2.0.0])
         AC_PREPROC_IFELSE([AC_LANG_PROGRAM([[
+#include "xorg-server.h"
 #include "exa.h"
 #if EXA_VERSION_MAJOR < 2
 #error OLD EXA!
-- 
cgit v0.10.2

